skimage/transform/_geometric.py

```diff
@@ -549,9 +549,10 @@ def _apply_mat(self, coords, matrix):
         src = np.vstack((x, y, np.ones_like(x)))
         dst = src.T @ matrix.T
 
+        # avoid division by zero
+        dst[dst[:, 2] == 0, 2] = np.finfo(float).eps
         # rescale to homogeneous coordinates
-        dst[:, 0] /= dst[:, 2]
-        dst[:, 1] /= dst[:, 2]
+        dst[:, :2] /= dst[:, 2:3]
```

Comment:
why not ``dst[:, :2] /= dst[:, 2]``? Broadcasting should work here I think?

Reply:
@emmanuelle Not really. `dst[:, :2]` dimension is `(n, 2)`, `dst[:, 2:3]` has a dimension of `(n,1)`, while `dst[:, 2]` dimension is just `n`.